### ISSUE TITLE
[CLOSED] Fixed Virtual server state enabled/disabled issue

### ISSUE THREAD

**Issue by RavinderReddyF5** [RavinderReddyF5/terraform-provider-bigip-version0.12#184]
<a href="https://github.com/RavinderReddyF5"><img src="https://avatars1.githubusercontent.com/u/45624469?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [RavinderReddyF5](https://github.com/RavinderReddyF5)**
_Friday Aug 23, 2019 at 07:22 GMT_
_Originally opened as https://github.com/terraform-providers/terraform-provider-bigip/pull/153_

----

----

_**[RavinderReddyF5](https://github.com/RavinderReddyF5)** included the following code: https://github.com/terraform-providers/terraform-provider-bigip/pull/153/commits_

**Comment by RavinderReddyF5** [comment]
<a href="https://github.com/scshitole"><img src="https://avatars2.githubusercontent.com/u/13858248?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [scshitole](https://github.com/scshitole)**
_Monday Aug 26, 2019 at 17:10 GMT_

----

@dannyk81 @RavinderReddyF5 is it ok to merge now

**Comment by RavinderReddyF5** [comment]
<a href="https://github.com/dannyk81"><img src="https://avatars0.githubusercontent.com/u/24274156?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [dannyk81](https://github.com/dannyk81)**
_Monday Aug 26, 2019 at 17:19 GMT_

----

@scshitole none of my comments were addressed, so no-go from me :smile: but I'm not a maintainer here, so ultimately not my decision.

**Comment by RavinderReddyF5** [comment]
<a href="https://github.com/dannyk81"><img src="https://avatars0.githubusercontent.com/u/24274156?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [dannyk81](https://github.com/dannyk81)**
_Monday Aug 26, 2019 at 17:46 GMT_

----

On a more general note, I feel like the review/CI/merge/release process here is not up to par with other officially hosted Hashicorp providers.

I believe Hashicorp has a CI framework (in Teamcity) in parallel to the Travis CI that can be used to perform unit & e2e tests and F5 can probably provide a hosted VE bigip box for running tests against.

Currently there's very little being tested on a regular basis and a contributor needs a Bigip box to actually test things.

increasing the pool of maintainers/reviewers/approvers can probably help in this regard as well.

again, just my 2 cents :smile:

**Comment by RavinderReddyF5** [comment]
<a href="https://github.com/scshitole"><img src="https://avatars2.githubusercontent.com/u/13858248?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [scshitole](https://github.com/scshitole)**
_Monday Aug 26, 2019 at 18:44 GMT_

----

@dannyk81 thanks we already have Team City CI Framework in place, thats how the binaries are released currently

**Comment by RavinderReddyF5** [comment]
<a href="https://github.com/dannyk81"><img src="https://avatars0.githubusercontent.com/u/24274156?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [dannyk81](https://github.com/dannyk81)**
_Monday Aug 26, 2019 at 19:22 GMT_

----

@scshitole is it running the full suite of tests? and a Travis based test suite for (at least) approved PRs seems like a good thing to have, so we can verify we don't have any regressions *prior* to merging.

From I've seen until now, full test suite is only executed manually sparingly which results in reverts and confusion.

This PR is a good example, it adds tests but we don't have any visibility on their results.

**Comment by RavinderReddyF5** [comment]
<a href="https://github.com/RavinderReddyF5"><img src="https://avatars1.githubusercontent.com/u/45624469?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [RavinderReddyF5](https://github.com/RavinderReddyF5)**
_Tuesday Aug 27, 2019 at 06:17 GMT_

----

@dannyk81 
we are running full regression test for Every Merge Request to make sure it will not break anything.

**Comment by RavinderReddyF5** [comment]
<a href="https://github.com/dannyk81"><img src="https://avatars0.githubusercontent.com/u/24274156?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [dannyk81](https://github.com/dannyk81)**
_Tuesday Aug 27, 2019 at 13:08 GMT_

----

I assume these tests are executed manually, since it's not part of the CI (at least from what I see, travis runs `make test` which doesn't set `TF_ACC=1`)

but anyway, we digress... could you refer to the comments I left regarding this PR?

**Comment by RavinderReddyF5** [comment]
<a href="https://github.com/scshitole"><img src="https://avatars2.githubusercontent.com/u/13858248?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [scshitole](https://github.com/scshitole)**
_Thursday Sep 05, 2019 at 16:46 GMT_

----

@RavinderReddyF5 @dannyk81 are we good to commit this PR ?

**Comment by RavinderReddyF5** [comment]
<a href="https://github.com/dannyk81"><img src="https://avatars0.githubusercontent.com/u/24274156?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [dannyk81](https://github.com/dannyk81)**
_Thursday Sep 05, 2019 at 16:55 GMT_

----

I don't intend to block this..
